sentry-core/src/client.rs

```diff
@@ -57,12 +68,46 @@ impl Client {
     ///
     /// If the DSN on the options is set to `None` the client will be entirely
     /// disabled.
-    pub fn with_options(options: ClientOptions) -> Client {
+    pub fn with_options(mut options: ClientOptions) -> Client {
         let transport = RwLock::new(match options.dsn {
             Some(_) => Some(Arc::new(options.transport.create_transport(&options))),
             None => None,
         });
-        Client { options, transport }
+
+        let integrations: HashMap<_, _> = options
+            .integrations
+            .iter()
+            .map(|integration| (integration.as_ref().type_id(), integration.clone()))
+            .collect();
+
+        let mut global = GLOBAL_INTEGRATIONS.lock().unwrap();
+        for (id, integration) in integrations.iter() {
+            if !global.contains(&id) {
+                global.insert(id.clone());
+                integration.setup_once();
+            }
+            integration.setup(&mut options);
+        }
+
+        Client {
+            options,
+            transport,
+            integrations: RwLock::new(integrations),
+        }
+    }
+
+    pub(crate) fn get_integration<I>(&self) -> Option<&I>
+    where
+        I: Integration,
+    {
+        let integrations = self.integrations.read().unwrap();
+        let integration = integrations.get(&TypeId::of::<I>())?;
+        // We can‘t use the following code, because it requires a `'static`
+        // reference, and the reference has the same lifetime as `self`.
+        // (&integration.as_ref() as &dyn Any).downcast_ref()
+        // SAFETY: we just checked for the concrete type via TypeId
+        let integration = unsafe { &*(&**integration as *const dyn Integration as *const I) };
```

Comment:
Potential solution for this: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=7dc36186ce572191e5872717b18f898b